accounts/views.py

```diff
@@ -28,21 +29,22 @@ def edit_profile(request):
                 profile.pronoun_preference = pronoun_preference
             profile.save()
             updated_user = request.user
-            updated_pronoun_preference = profile.get_pronoun_preference_display() 
-            return render(request, 'profile_updated.html', {'user': updated_user, 'pronoun_preference': updated_pronoun_preference})
-            # return HttpResponseRedirect(reverse('profile_updated'))
-            # return render('profile_updated')
+            updated_pronoun_preference = profile.get_pronoun_preference_display()
+            return HttpResponseRedirect(reverse('profile_updated'))
+            #return render(request, 'profile_updated.html', {'user': updated_user, 'pronoun_preference': updated_pronoun_preference})
+
     else:
         form = EditProfileForm(instance=profile)
-        # pronoun_preference = profile.get_pronoun_preference_display() 
-        # return render(request, 'home.html', {'user': request.user, 'pronoun_preference': pronoun_preference, 'form': form})
+        return render(request, 'edit_profile.html', {'form': form})
 
-    # return render(request, 'edit_profile.html', {'form': form})
     pronoun_preference = profile.get_pronoun_preference_display()
     return render(request, 'home.html', {'user': request.user, 'pronoun_preference': pronoun_preference, 'form': form})
 
+
 def profile_updated(request):
-    updated_user = request.session.get('updated_user')
-    updated_pronoun_preference = request.session.get('updated_pronoun_preference')
+    profile = request.user.profile
```

Comment:
One of the key parts! We don't look at the sessions updated user, but at the actual changed profile in our DB.